Lib/ufo2ft/compilers/baseCompiler.py

```diff
@@ -0,0 +1,304 @@
+            callables.append(self.preProcessorClass.initDefaultFilters)
+        preprocessor_args = prune_unknown_kwargs(self.__dict__, *callables)
+        preProcessor = self.preProcessorClass(ufo_or_ufos, **preprocessor_args)
+        return preProcessor.process()
+
+    # @timer("compile a basic TTF")
+    def compileOutlines(self, ufo, glyphSet):
+        kwargs = prune_unknown_kwargs(self.__dict__, self.outlineCompilerClass)
+        kwargs["tables"] = self._tables
+        outlineCompiler = self.outlineCompilerClass(ufo, glyphSet=glyphSet, **kwargs)
+        return outlineCompiler.compile()
+
+    # @timer("postprocess TTF"):
+    def postprocess(self, ttf, ufo, glyphSet):
+        if self.postProcessorClass is not None:
+            postProcessor = self.postProcessorClass(ttf, ufo, glyphSet=glyphSet)
+            kwargs = prune_unknown_kwargs(self.__dict__, postProcessor.process)
+            ttf = postProcessor.process(**kwargs)
+        return ttf
+
+    def compileFeatures(
+        self,
+        ufo,
+        ttFont=None,
+        glyphSet=None,
+    ):
+        """Compile OpenType Layout features from `ufo` into FontTools OTL tables.
+        If `ttFont` is None, a new TTFont object is created containing the new
+        tables, else the provided `ttFont` is updated with the new tables.
+
+        If no explicit `featureCompilerClass` is provided, the one used will
+        depend on whether the ufo contains any MTI feature files in its 'data'
+        directory (thus the `MTIFeatureCompiler` is used) or not (then the
+        default FeatureCompiler for Adobe FDK features is used).
+
+        If skipExportGlyphs is provided (see description in the ``compile*``
+        functions), the feature compiler will prune groups (removing them if empty)
+        and kerning of the UFO of these glyphs. The feature file is left untouched.
+
+        `debugFeatureFile` can be a file or file-like object opened in text mode,
+        in which to dump the text content of the feature file, useful for debugging
+        auto-generated OpenType features like kern, mark, mkmk etc.
+        """
+        if self.featureCompilerClass is None:
+            if any(
+                fn.startswith(MTI_FEATURES_PREFIX) and fn.endswith(".mti")
+                for fn in ufo.data.fileNames
+            ):
+                self.featureCompilerClass = MtiFeatureCompiler
+            else:
+                self.featureCompilerClass = FeatureCompiler
+
+        kwargs = prune_unknown_kwargs(self.__dict__, self.featureCompilerClass)
+        featureCompiler = self.featureCompilerClass(
+            ufo, ttFont, glyphSet=glyphSet, **kwargs
+        )
+        otFont = featureCompiler.compile()
+
+        if self.debugFeatureFile:
+            if hasattr(featureCompiler, "writeFeatures"):
+                featureCompiler.writeFeatures(self.debugFeatureFile)
+
+        return otFont
+
+
+class DesignspaceCompiler(BaseCompiler):
```

Comment:
I found the name of this class a bit unclear/ambiguous. Maybe call this BaseInterpolatableCompiler, to make clear this is a Base class not meant to be instantiated and it's the base class for the Interopolatable{TTF,OTF}Compiler classes